### PR TITLE
`get-counter` issue with `-computer` parameter

### PR DESCRIPTION
**Pull request by krishnayalavarthi**
# PR Summary

Get-Counter: fails when "-computer" parameter present

## PR Context

Assuming the local system is named MySystem, the following should
work:

get-counter -computer MySystem "\Processor(_Total)\% Processor Time"
This fails because get-counter inserts an extra backslash into the
string between "MySystem" and "\Processor", so the full path ends up
being \\MySystem\\Processor(_Total)\% Processor Time.

PDH then parses this path as follows:

Machine = MySystem\
CounterSet = Processor
Instance = _Total
Counter = % Processor Time
Note that the extra slash remains in Machine. As a result, PDH does not
recognize that this is a query for the local system, and instead
attempts to perform a remote query to the "MySystem" machine, which
does not exist. The query fails.

Fixes:

Update the Get-Counter cmdlet to not insert the extra backslash.
Update PDH to trim trailing backslashes from the machine name.

## PR Checklist

- [ ] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [ ] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Comment by krishnayalavarthi** [comment]
#15165

**Comment by iSazonov** [comment]
I don't see any failures:
```powershell
PowerShell 7.1.0
Copyright (c) Microsoft Corporation.

https://aka.ms/powershell
Type 'help' to get help.


PS C:\install\PowerShell\7> Get-Counter -Counter "\Processor(_Total)\% Processor Time"  -ComputerName localhost

Timestamp                 CounterSamples
---------                 --------------
4/6/2021 9:17:31 PM       \\localhost\\processor(_total)\% processor time :
                          10.159475274837638

**Review by TravisEz13 — DISMISSED** [review]

File: src/Microsoft.PowerShell.Commands.Diagnostics/GetCounterCommand.cs [hunk 1]
```diff
@@ -603,13 +603,14 @@ private List<string> CombineMachinesAndCounterPaths()
                 {
                     foreach (string machine in _computerName)
                     {
+                        string slashBeforePath = path.Length > 0 && path[0] == '\\' ? "" : "\\";
```
Comment (TravisEz13) [inline src/Microsoft.PowerShell.Commands.Diagnostics/GetCounterCommand.cs:606]:
```suggestion
                        string slashBeforePath = path.Length > 0 && path[0] == '\\' ? string.Empty : "\\";
```

**Comment by TravisEz13** [comment]
@iSazonov I think it has special code for localhost, try the actual machine name.

**Comment by krishnayalavarthi** [comment]
@TravisEz13 Looks fine. You can approve.

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**